lib/components/user/trip-summary.js

```diff
@@ -0,0 +1,85 @@
+import coreUtils from '@opentripplanner/core-utils'
+import { ClassicLegIcon } from '@opentripplanner/icons'
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+
+import { getQueryParamsFromQueryString } from '../../util/query'
+import ItinerarySummary from '../narrative/default/itinerary-summary'
+
+const { formatDuration, formatTime } = coreUtils.time
+
+class TripSummary extends Component {
+  constructor () {
+    super()
+
+    this.state = {
+      // Hold a default state for query location names
+      // while it is being fetched in updateQueryState.
+      queryObj: {
+        from: {},
+        to: {}
+      }
+    }
+  }
+
+  updateQueryState = async () => {
+    const { monitoredTrip, config } = this.props
+    const { queryParams } = monitoredTrip
+    const queryObj = await getQueryParamsFromQueryString(queryParams, config)
+
+    // Prevent a state update if componentWillUnmount happened while await was executing.
+    if (!this.state.willUnmount) {
+      this.setState({ queryObj })
+    }
+  }
+
+  componentDidMount () {
+    this.updateQueryState()
+  }
+
+  componentDidUpdate (prevProps) {
+    // Update the state the monitoredTrip prop has changed.
+    if (this.props.monitoredTrip !== prevProps.monitoredTrip) {
+      this.updateQueryState()
+    }
+  }
+
+  componentWillMount () {
+    this.setState({
+      willUnmount: true
```

Comment:
This state update and updateQueryState needs a lot more comments about it does. On first blush, it doesn't really seem to be doing things in a standard way. Perhaps the `getQueryParamsFromQueryString` should just be called in `mapStateToProps` to avoid some of the weirdness with state updates?

Reply:
Turns out tracking the mounted state is an antipattern: https://reactjs.org/blog/2015/12/16/ismounted-antipattern.html. I found this https://juliangaramendy.dev/use-promise-subscription/, and a basic form of the Promise handling could do the trick. `getQueryParamsFromQueryString` is not an action, so redux should not be involved in here.

Reply:
https://github.com/opentripplanner/otp-react-redux/pull/174/commits/ad8e678e36fec4268c8bdbbb4c29f585aedf4c75